### ISSUE TITLE
chokidar add file triggered too fast. 

### ISSUE THREAD

**Issue by eshengit** [paulmillr/chokidar#453]
Its similar issue like this but the workaround in this post does not work for me. 
http://stackoverflow.com/questions/34750158/chokidar-onchange-event-for-a-file-is-possibly-triggered-to-fast

**the setting:**
var watcher = chokidar.watch('file', {
    ignored: '*.tgz',
    alwaysStat: true,
    ignoreInitial: true,
    usePolling: true,
    awaitWriteFinish: {
         stabilityThreshold: 3000,
         pollInterval: 100
    }
});
watcher.add(path)
watcher
    .on('add', function (path) {
        handler.handle(path)
    })

**the problem:**
in the handler class which is fired only when the file is completely written(expected behavior)
if i run ls -l $newfile, I can see most of the time, the file is zero size or partially written. 

my os is redhat. .8.13-16.2.1.el6uek.x86_64
I am not sure it happens because the os cache the writing and flush only on batch?
the new file is generated by $somecmd > $newfile

**Comment by es128** [comment]
You're running `ls -l $newfile` from within your node script somehow or at the terminal? If the latter, I'm not sure chokidar has anything to do with whatever is going on.

Does it happen if you watch the file from the outset instead of `.add()`ing it later? Like `chokidar.watch(path, {...})`.

**Comment by eshengit** [comment]
@es128,  I did some file operation inside handler.handle. 
this file op made me realized that the file content is not ready so I added ls -l $newfile to check if my assumption is correct.  

watcher
.on('add', function (path) {
handler.handle(path)
})

I also have a question: 
is the watcher.on('add', functionA),  is the listener function executed in asyn or synced way?  
if each function take long time to execute, and new file generated very fast, could we have couple function A running at the same time, or we have one function A running at any given moment. 
I am new to node,:)
